ql/src/java/org/apache/hadoop/hive/ql/io/arrow/Serializer.java

```diff
@@ -347,6 +347,21 @@ private void writeStruct(NonNullableStructVector arrowVector, StructColumnVector
     final ColumnVector[] hiveFieldVectors = hiveVector == null ? null : hiveVector.fields;
     final int fieldSize = fieldTypeInfos.size();
 
+    // This is to handle following scenario -
+    // if any struct value itself is NULL, we get structVector.isNull[i]=true
+    // but we don't get the same for it's child fields which later causes exceptions while setting to arrow vectors
+    // see - https://issues.apache.org/jira/browse/HIVE-25243
+    if (hiveVector != null && hiveFieldVectors != null) {
+      for (int i = 0; i < size; i++) {
+        if (hiveVector.isNull[i]) {
+          for (ColumnVector fieldVector : hiveFieldVectors) {
+            fieldVector.isNull[i] = true;
```

Comment:
Shouldn't it set only specific filed vector to null. For every i, why its iterating over all the hiveFieldVectors.

Reply:
It is like - if for outer vector(struct here) isNull[i] = true, all the nested vectors (hiveFieldVectors) should also have isNull[i] = true.

Reply:
to me it looks like if one of the filed is null ..then all fields are set to null.